backend/capellacollab/projects/toolmodels/restrictions/routes.py

```diff
@@ -48,7 +48,7 @@ def update_restrictions(
 ) -> models.DatabaseToolModelRestrictions:
     if body.allow_pure_variants and not model.tool.integrations.pure_variants:
         raise fastapi.HTTPException(
-            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            status_code=status.HTTP_409_CONFLICT,
```

Comment:
I think bad request is better here as the client does a request referencing a model with an invalid tool configuration.